common/hexutil/hexutil_test.go

```diff
@@ -28,9 +28,10 @@ type marshalTest struct {
 }
 
 type unmarshalTest struct {
-	input   string
-	want    interface{}
-	wantErr error
+	input        string
+	want         interface{}
+	wantErr      error // if set, decoding must fail on any platform
+	wantErr32bit error // if set, decoding must fail on 32bit platforms (used for Uint tests)
```

Comment:
I'm not entirely sure about this. Given that there may be errors working with int/uint, why support it at all? Seems to me to be just a fancy use case that will bite people.